File: src/Editors/Editors Lib/src/file/NoCoDec.java
package file;


/**
 * Codification class that does not encode/decode data
 * @author Gil Costa
 */
public class NoCoDec extends ACoDec implements ICoDec{

	@Override
	public byte[] encode(byte[] data) {
		return data;
	}
	
	
	@Override
	public byte[] decode(byte[] data) {
		return data;
	}
}

File: src/Engine/Files/NoCoDec.cpp

#include "NoCodec.h"
#include <fstream>


/**
 * encode method
 * @param data the byteArray to encode
 * @return the result of encoding the given byteArray
 */
Data NoCoDec::encode(const Data& data) throw(){
    // Does nothing
    return data;
}
/**
 * decode method
 * @param data the byteArray to decode
 * @return the result of decoding the given byteArray
 */
Data NoCoDec::decode(const Data& data) throw(){
    // does nothing
    return data;
}
